opentelemetry-zipkin/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+}
+
+impl Default for ExporterConfigBuilder {
+    fn default() -> Self {
+        ExporterConfigBuilder {
+            collector_endpoint: None,
+            service_name: None,
+            service_endpoint: None,
+        }
+    }
+}
+
+impl ExporterConfig {
+    pub fn builder() -> ExporterConfigBuilder {
+        ExporterConfigBuilder::default()
+    }
+}
+
+impl ExporterConfigBuilder {
+    /// Create `ExporterConfig` struct from current `ExporterConfigBuilder`
+    pub fn build(&self) -> ExporterConfig {
+        let local_endpoint: endpoint::Endpoint;
+        let service_name = self
+            .service_name
+            .clone()
+            .unwrap_or_else(|| "DEFAULT".to_owned());
+        match self.service_endpoint {
+            Some(socket_addr) => {
+                match socket_addr.ip() {
+                    net::IpAddr::V4(addr) => {
+                        local_endpoint = endpoint::Endpoint::builder()
+                            .service_name(service_name)
+                            .ipv4(addr)
+                            .port(socket_addr.port())
+                            .build()
+                    }
+                    net::IpAddr::V6(addr) => {
+                        local_endpoint = endpoint::Endpoint::builder()
+                            .service_name(service_name)
+                            .ipv6(addr)
+                            .port(socket_addr.port())
+                            .build()
+                    }
+                };
+            }
+            None => {
+                local_endpoint = endpoint::Endpoint::builder()
+                    .service_name(service_name)
+                    .build()
+            }
+        };
+        ExporterConfig {
+            collector_endpoint: self
+                .collector_endpoint
+                .clone()
+                .unwrap_or_else(|| DEFAULT_COLLECTOR_ENDPOINT.parse().unwrap()),
+            local_endpoint,
+        }
+    }
+
+    /// Assign the service name for `ConfigBuilder`
+    pub fn with_service_name(&mut self, name: String) -> &mut Self {
+        self.service_name = Some(name);
+        self
+    }
+
+    /// Assign the service endpoint for `ConfigBuilder`
+    pub fn with_service_endpoint(&mut self, endpoint: net::SocketAddr) -> &mut Self {
+        self.service_endpoint = Some(endpoint);
+        self
+    }
+}
+
+impl Exporter {
+    /// Creates new `Exporter` from a given `ExporterConfig`.
+    pub fn from_config(config: ExporterConfig) -> Self {
+        Exporter {
+            config: config.clone(),
+            uploader: Mutex::new(uploader::Uploader::new(
+                config.collector_endpoint,
+                uploader::UploaderFormat::HTTP,
+            )),
+        }
+    }
+}
+
+impl trace::SpanExporter for Exporter {
+    /// Export spans to Zipkin collector.
+    fn export(&self, batch: Vec<Arc<trace::SpanData>>) -> trace::ExportResult {
+        match self.uploader.lock() {
```

Comment:
I don't think you need this Mutex (it is needed by the jaeger exporter because the thrift api requires mutable references, but reqwest does not) the functions in uploader could just take `&self`.